src/assignment/assignment.service.ts

```diff
@@ -509,21 +516,36 @@ export class AssignmentService {
 	static async duplicateAssignment(
 		newTitle: string,
 		initialAssignment: Partial<Avo.Assignment.Assignment> | null,
-		profileId: string
+		user: UserSchema
 	): Promise<Avo.Assignment.Assignment> {
-		if (!initialAssignment || !initialAssignment.id) {
+		const owner_profile_id = user.profile?.id;
```

Comment:
The code should use camelcase for variables
only switch to snake case at the last moment just before interacting with the database

```suggestion
		const ownerProfileId = user.profile?.id;
```